internal/cmd/config.go

```diff
@@ -2263,24 +2267,26 @@ func (c *Config) newTemplateData(cmd *cobra.Command) *templateData {
 	sourceDirAbsPath, _ := c.getSourceDirAbsPath(nil)
 
 	return &templateData{
-		arch:         runtime.GOARCH,
-		args:         os.Args,
-		cacheDir:     c.CacheDirAbsPath,
-		command:      cmd.Name(),
-		config:       c.ConfigFile.toMap(),
-		configFile:   c.configFileAbsPath,
-		executable:   chezmoi.NewAbsPath(executable),
-		fqdnHostname: fqdnHostname,
-		gid:          gid,
-		group:        group,
-		homeDir:      c.homeDirAbsPath,
-		hostname:     hostname,
-		kernel:       kernel,
-		os:           runtime.GOOS,
-		osRelease:    osRelease,
-		sourceDir:    sourceDirAbsPath,
-		uid:          uid,
-		username:     username,
+		arch:              runtime.GOARCH,
+		args:              os.Args,
+		cacheDir:          c.CacheDirAbsPath,
+		command:           cmd.Name(),
+		config:            c.ConfigFile.toMap(),
+		configFile:        c.configFileAbsPath,
+		executable:        chezmoi.NewAbsPath(executable),
+		fqdnHostname:      fqdnHostname,
+		gid:               gid,
+		group:             group,
+		homeDir:           c.homeDirAbsPath,
+		hostname:          hostname,
+		kernel:            kernel,
+		os:                runtime.GOOS,
+		osRelease:         osRelease,
+		pathListSeparator: string(os.PathListSeparator),
+		pathSeparator:     string(os.PathSeparator),
```

Comment:
The typecasts to `string` are not needed here as the values are already strings.

Reply:
In go 1.21 they are defined as runes.

https://github.com/golang/go/blob/3fb1d95149fa280343581a48547c3c3f70dac5fb/src/os/path_unix.go#L5-L13

They test as int32 too:

```golang
package psstr

import (
	"os"
	"reflect"
	"testing"
)

func TestName(t *testing.T) {
	to := reflect.TypeOf(os.PathSeparator)
	expected := "string"
	if to.String() != expected {
		t.Fatalf("PathSeparator is %s not %s", to.String(), expected)
	}
}
```

```console
=== RUN   TestOne
    main_test.go:13: PathSeparator is int32 not string
--- FAIL: TestOne (0.00s)

```

I can't test direct comparison:
```golang
func TestTwo(t *testing.T) {
	ps := os.PathSeparator
	expected := "/"
	if ps != expected {
		t.Fatalf("PathSeparator %s is not equal to %s", ps, expected)
	}
}
```
I get
```console
# psstr [psstr.test]
./main_test.go:20:11: invalid operation: ps != expected (mismatched types rune and string)

```

Reply:
It doesn't look like the `os` package is being shadowed either, it's definitely being imported:
https://github.com/twpayne/chezmoi/blob/d637ddddaa8484622aa36e39eb6fe33d779bbfb3/internal/cmd/config.go#L12-L14

Reply:
I'm sorry, I was wrong. For some reason I was sure that they were constant strings, but you're right, they're runes.

Reply:
Yeah I've made the same mistake in the past. I guess it's a safe assumption that there aren't any multi-rune path separator OSes in common usage. I feel there is some weirdism in go around conversion between "primitive" / "native" types. Given everywhere else you have to be rather explicit